### PR TITLE
MULTIARCH-4696: change ic to ibmcloud, remove unused terraform/golang branch

### PR DESCRIPTION
**Pull request by prb112**
- install vpc/powervs converted to ic
- cleaned up code base
- cluster_type was changed unintentionally and did not match

### CODE REVIEW

**Review by pkenchap — APPROVED** [review]

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh [hunk 1]
```diff
@@ -1,37 +1,50 @@
...
 
-# Generates a workspace name like rdr-multi-arch-4-14-au-syd-n1
-# this keeps the workspace unique per version
-OCP_VERSION=$(cat "${SHARED_DIR}/OCP_VERSION")
-REGION="${LEASED_RESOURCE}"
-CLEAN_VERSION=$(echo "${OCP_VERSION}" | tr '.' '-')
-if [ ! -z "${WORKFLOW_TYPE}" ]
+##### Constants
+IBMCLOUD_HOME=/tmp/ibmcloud
+export IBMCLOUD_HOME
```
Comment (tvardema) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh:5]: How is this variable used?  I don't see "`IBMCLOUD_HOME`" used in this script at all, but is it used from the environment variables for commands executed or something?

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
+
+# Variables
+# OCP Version
+if [ ! -f "${SHARED_DIR}/OCP_VERSION" ]
 then
-  CUCUSHIFT="-cucushift"
-else
-  CUCUSHIFT=""
+    echo "Provisioning never happened"
+    exit 0
```
Comment (tvardema) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh:17]: If provisioning never happens, wouldn't we want to report failure, not success, from this step?

**Review by tvardema — COMMENTED** [review]
Possible functional comment, otherwise it looks good!

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh (hunk 2 above)
Comment (prb112) [reply ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh:17]: it's deprovision

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh (hunk 1 above)
Comment (prb112) [reply ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh:5]: it's pickedup by the CLI

Comment (tvardema) [reply ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh:5]: Answered async in slack.  Thanks!

**Pull request merged**